### PR TITLE
Use the configuration class for setting the directories

### PR DESCRIPTION
**Pull request by teclator**
- It uses the `Yast::ConfigurationManagement::Configurations::Base class` for the directories configuration.
- It moves all the formula handling into the `Formula Sequence`

### CODE REVIEW

File: src/lib/configuration_management/configurations/salt.rb [hunk 1]
```diff
@@ -21,6 +22,9 @@ def post_initialize(options)
           @type       = "salt"
           @states_url = URI(options[:states_url]) if options[:states_url]
           @pillar_url = URI(options[:pillar_url]) if options[:pillar_url]
+          @custom_pillar_root = pathnames_from(options[:pillar_root])
```
[the remark this answers is not in the document]
Comment (teclator) [reply src/lib/configuration_management/configurations/salt.rb]: It should be a `Pathname`

File: src/lib/configuration_management/configurations/salt.rb [hunk 2]
```diff
@@ -43,6 +48,29 @@ def pillar_root(scope = :local)
         def formulas_root(scope = :local)
           work_dir(scope).join("formulas")
         end
+
+        # Return paths to the states root
```
[the remark this answers is not in the document]
Comment (teclator) [reply src/lib/configuration_management/configurations/salt.rb]: I will improve the documentation, just cherry picked this changes for using the variables. But the main idea of the PR was about discussing where we should put the responsibility of reading and writing the pillar data, and also writing the states files.

File: src/lib/y2configuration_management/clients/main.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      end
+
+    private
+
+      # Returns the configuration management configuration
+      #
+      # @return [Yast::ConfigurationManagement::Configurations::Base]
+      def config
+        return @config if @config
+        settings =
```
Comment (mvidner) [inline src/lib/y2configuration_management/clients/main.rb]: where is the type of `settings` documented?

File: src/desktop/configuration_management.desktop [hunk 4]
```diff
@@ -13,6 +13,6 @@ X-SuSE-YaST-AutoInstClient=configuration_management_auto
 X-SuSE-YaST-Keywords=salt,privisioning,configuration_management
 
 Icon=yast2-software
-Exec=/usr/bin/xdg-su -c '/sbin/yast2 configuration_management'
+Exec==/usr/bin/xdg-su -c "/sbin/yast2 configuration_management"
```
Comment (mvidner) [inline src/desktop/configuration_management.desktop]: Double equal sign is intentional two?

Comment (teclator) [reply src/desktop/configuration_management.desktop]: Nop, that was an error when editing. Thnx

**Comment by mvidner** [comment]
OK, so what are the user visible changes that I should expect/test in this PR?

**Comment by teclator** [comment]
@mvidner 

From the user POV the main change is about running Salt.

https://github.com/yast/yast-configuration-management/pull/27/files#diff-cc0c8f6af7bf634e526af007e96f2dceR69

But the main purpose of the PR was about unifying the way the configuration was read / applied and about moving that responsibility from the client to the already existent `Configurator`.

So basically trying to not repeat ourselves and sharing as much as possible with the firstboot client.

**Comment by mvidner** [comment]
When I run the configuration_management client and click Abort in the formula selection dialog, I get the same dialog again in an endless loop.

**Comment by mvidner** [comment]
I've run the module, intending to apply only the TFTP formula. To my surprise, also other formulas were applied, because a previous run(?) left some other states in `/srv/susemanager/formula_data/pillar/top.sls` :frowning_face:

File: src/lib/y2configuration_management/salt/formula.rb [hunk 5]
```diff
@@ -90,14 +90,21 @@ def description
 
       # Return all the installed formulas
       #
+      # @note The result is cached. To force refreshing the cache, set the `reload`
+      #   parameter to `true`.
+      #
+      # @param paths  [Array<String>|String] File system paths to search for formulas
```
Comment (mvidner) [inline src/lib/y2configuration_management/salt/formula.rb]: If we use the `*paths` splat then a Single string is impossible

**Pull request merged**